menu_launcher.py

```diff
@@ -624,8 +628,20 @@ def run_plugins(path_dirs, action):
         for plug in plugin_array:
             plugins[plug] = config.get("plugins", plug)
 
+        # get number of installed cores
+        cores = len(get_installed_cores(path_dirs))
+
+        # get number of installed collectors
+        all_colls = len(get_installed_collectors(path_dirs, "all"))
+        passive_colls = len(get_installed_collectors(path_dirs, "passive"))
+        active_colls = len(get_installed_collectors(path_dirs, "active"))
+
+        # get number of installed visualizations
+        vis = len(get_installed_vis(path_dirs))
+
         for plugin in plugins:
-            if plugin in ["core", "visualization"]:
+            # check if plugin is core or vis and the corresponding size is greater than 0
+            if (plugin, cores > 0) in [("core", True)] or (plugin, vis > 0) in [("visualization", True)]:
```

Comment:
I believe so, cores only measures all installed cores, whereas the visualization namespace has to be considered. I'm checking for strict ordering of plugin == "core" and cores > 0 or plugin == "visualization" and vis > 0. The above way would only check if we had enough cores.